src/converters/app-converter.ts

```diff
@@ -0,0 +1,388 @@
+            throw new Error(`missing CDK Stack for artifact ${artifact.id}`);
+        }
+        stackConverter.convert(dependencies);
+    }
+}
+
+/**
+ * StackConverter converts all of the resources in a CDK stack to Pulumi resources
+ */
+export class StackConverter extends ArtifactConverter {
+    readonly parameters = new Map<string, any>();
+    readonly resources = new Map<string, Mapping<pulumi.Resource>>();
+    readonly constructs = new Map<ConstructInfo, pulumi.Resource>();
+    stackResource!: CdkConstruct;
+
+    constructor(host: IStackComponent, readonly stack: StackManifest) {
+        super(host);
+    }
+
+    public convert(dependencies: Set<ArtifactConverter>) {
+        const dependencyGraphNodes = GraphBuilder.build(this.stack);
+
+        // process parameters first because resources will reference them
+        for (const [logicalId, value] of Object.entries(this.stack.parameters ?? {})) {
+            this.mapParameter(logicalId, value.Type, value.Default);
+        }
+
+        for (const n of dependencyGraphNodes) {
+            if (n.construct.id === this.stack.id) {
+                this.stackResource = new CdkConstruct(`${this.host.name}/${n.construct.path}`, n.construct.id, {
+                    parent: this.host,
+                    // TODO: we could do better here. Currently the stack depends on all assets, but really
+                    // only individual resources should depend on individual assets
+                    dependsOn: this.stackDependsOn(dependencies),
+                });
+                this.constructs.set(n.construct, this.stackResource);
+                continue;
+            }
+
+            const parent = this.constructs.get(n.construct.parent!)!;
+            if (n.resource && n.logicalId) {
+                const cfn = n.resource;
+                debug(`Processing node with template: ${JSON.stringify(cfn)}`);
+                debug(`Creating resource for ${n.logicalId}`);
+                const props = this.processIntrinsics(cfn.Properties);
+                const options = this.processOptions(cfn, parent);
+
+                const mapped = this.mapResource(n.logicalId, cfn.Type, props, options);
+                const resource = pulumi.Resource.isInstance(mapped) ? mapped : mapped.resource;
+                const attributes = pulumi.Resource.isInstance(mapped) ? undefined : mapped.attributes;
+                this.resources.set(n.logicalId, { resource, attributes, resourceType: cfn.Type });
+                this.constructs.set(n.construct, resource);
+
+                debug(`Done creating resource for ${n.logicalId}`);
+                // TODO: process template conditions
+                // for (const [conditionId, condition] of Object.entries(cfn.Conditions || {})) {
+                //     // Do something with the condition
+                // }
+            } else {
+                const r = new CdkConstruct(`${this.host.name}/${n.construct.path}`, n.construct.type, {
```

Comment:
Might be worth extracting this pattern into a function. I saw it in another place as well

Reply:
I think it is extracted as much as possible, the inputs to `CdkConstruct` are slightly different between the two.